modules/oe_translation_poetry/src/Poetry.php

```diff
@@ -90,29 +139,33 @@ public function __construct(array $settings, ConfigFactoryInterface $configFacto
       'service.password' => Settings::get('poetry.service.password'),
       'notification.username' => Settings::get('poetry.notification.username'),
       'notification.password' => Settings::get('poetry.notification.password'),
-      'logger' => $loggerChannel,
+      'notification.endpoint' => Url::fromRoute('oe_translation_poetry.notifications')->setAbsolute()->toString(),
+      'logger' => $this->loggerChannel,
       'log_level' => LogLevel::INFO,
     ];
 
-    if (isset($settings['service_wsdl'])) {
-      $values['service.wsdl'] = $settings['service_wsdl'];
-    }
-
-    if (!isset($settings['site_id'])) {
-      $settings['site_id'] = $configFactory->get('system.site')->get('name');
+    if (isset($this->translatorSettings['service_wsdl'])) {
+      $values['service.wsdl'] = $this->translatorSettings['service_wsdl'];
     }
 
-    parent::__construct($values);
+    $this->poetryClient = new PoetryLibrary($values);
 
     // Register our own service provider to override the SOAP client. We need
     // to pass the current cookies to that the SOAP client uses them in its
     // requests. This is critical for ensuring the tests work.
-    $this->register(new PoetrySoapProvider($requestStack->getCurrentRequest()->cookies->all()));
+    $this->poetryClient->register(new PoetrySoapProvider($this->requestStack->getCurrentRequest()->cookies->all()));
+  }
 
-    $this->translatorSettings = $settings;
-    $this->state = $state;
-    $this->entityTypeManager = $entityTypeManager;
-    $this->database = $database;
+  /**
+   * Delegates to the Poetry library all calls made to this service.
+   *
+   * {@inheritdoc}
+   */
+  public function __call($name, $arguments) {
+    $this->initialize();
+    if (method_exists($this->poetryClient, $name)) {
+      return call_user_func_array([$this->poetryClient, $name], $arguments);
+    }
```

Comment:
And if method is not defined...? Shouldn't it throw an error?

Reply:
yes, just as if you call a missing method on any class.